cdc/owner.go

```diff
@@ -295,10 +296,19 @@ func (o *Owner) loadChangeFeeds(ctx context.Context) error {
 
 func (o *Owner) flushChangeFeedInfos(ctx context.Context) error {
 	snapshot := make(map[model.ChangeFeedID]*model.ChangeFeedStatus, len(o.changeFeeds))
+	minCheckpointTs := uint64(math.MaxUint64)
 	for id, changefeed := range o.changeFeeds {
 		snapshot[id] = changefeed.status
+		if changefeed.status.CheckpointTs < minCheckpointTs {
+			minCheckpointTs = changefeed.status.CheckpointTs
+		}
+	}
+	err := o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot)
+	if err != nil {
+		return errors.Trace(err)
 	}
-	return errors.Trace(o.cfRWriter.PutAllChangeFeedStatus(ctx, snapshot))
+	_,err = o.pdClient.UpdateGCSafePoint(ctx, minCheckpointTs)
```

Comment:
I think we should call [`UpdateServiceGCSafePoint`](https://github.com/pingcap/pd/blob/7bef70bee8a9fc59af741576328de0a4ecc15302/client/client.go#L74) ? @MyonKeminta